benchmarks/dynamo/pr_time_benchmarks/check_results.py

```diff
@@ -0,0 +1,141 @@
+    expected_data: dict[str, ExpectedFileEntry] = {}
+
+    with open(expected_file_path) as f:
+        reader = csv.reader(f)
+        for row in reader:
+            entry = ExpectedFileEntry(
+                benchmark_name=row[0].strip(),
+                metric_name=row[1].strip(),
+                expected_value=int(row[2]),
+                noise_margin=float(row[3]),
+            )
+            key = (entry.benchmark_name, entry.metric_name)
+            assert key not in expected_data, f"Duplicate entry for {key}"
+            expected_data[key] = entry
+
+    # Read result data file.
+    result_data: dict[str, ResultFileEntry] = {}
+
+    with open(result_file_path) as f:
+        reader = csv.reader(f)
+        for row in reader:
+            entry = ResultFileEntry(
+                benchmark_name=row[0].strip(),
+                metric_name=row[1].strip(),
+                actual_value=int(row[2]),
+            )
+
+            key = (entry.benchmark_name, entry.metric_name)
+            assert key not in result_data, f"Duplicate entry for {key}"
+            result_data[key] = entry
+
+    fail = False
+    for key, entry in expected_data.items():
+        if key not in result_data:
+            print(f"Missing entry for {key} in result file")
+            sys.exit(1)
+
+        low = entry.expected_value - entry.expected_value * entry.noise_margin
+        high = entry.expected_value + entry.expected_value * entry.noise_margin
+        result = result_data[key].actual_value
+
+        def log(event_name):
+            scribe.open_source_signpost(
+                subsystem="pr_time_benchmarks",
+                name=event_name,
+                parameters=json.dumps(
+                    {
+                        "benchmark_name": entry.benchmark_name,
+                        "metric_name": entry.metric_name,
+                        "actual_value": result,
+                        "expected_value": entry.expected_value,
+                        "noise_margin": entry.noise_margin,
+                    }
+                ),
+            )
+
+        if result > high:
+            fail = True
+            ratio = (float)(result - entry.expected_value) * 100 / entry.expected_value
+            print(
+                f"REGRESSION: benchmark {key} failed, actual result {result} "
+                f"is {ratio:.2f}% higher than expected {entry.expected_value} ±{entry.noise_margin*100:.2f}% "
+                f"if this is an expected regression, please update the expected results."
+            )
+
+            log("fail_regression")
+
+        if result < low:
+            fail = True
+            ratio = (float)(entry.expected_value - result) * 100 / entry.expected_value
+
+            print(
+                f"WIN: benchmark {key} failed, actual result {result} is {ratio:.2f}% lower than "
+                f"expected {entry.expected_value} ±{entry.noise_margin*100:.2f}% "
+                f"please update the expected results."
```

Comment:
OK, so are you going to write the updater script too?

Reply:
not in this diff